airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -0,0 +1,257 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+import os
+import urllib
+from pathlib import Path
+from typing import Any, Mapping, Optional, Tuple, Union
+
+import requests
+import requests_cache
+from airbyte_cdk.models import FailureType
+from airbyte_cdk.sources.http_config import MAX_CONNECTION_POOL_SIZE
+from airbyte_cdk.sources.streams.call_rate import APIBudget, CachedLimiterSession, LimiterSession
+from airbyte_cdk.utils.constants import ENV_REQUEST_CACHE_PATH
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+from requests.auth import AuthBase
+
+from .decoders import Decoder, JsonDecoder
+from .error_handlers import DefaultBackoffStrategy, HttpStatusErrorHandler, JsonErrorMessageParser, ResponseAction
+from .exceptions import DefaultBackoffException, RequestBodyException, UserDefinedBackoffException
+from .rate_limiting import default_backoff_handler, user_defined_backoff_handler
+
+BODY_REQUEST_METHODS = ("GET", "POST", "PUT", "PATCH")
+
+
+class HttpClient:
+    def __init__(
+        self,
+        stream_name: str,
```

Comment:
Oh! I forgot to ask: what about HTTP requests that are outside of the context of a stream. For example, authentication of dynamic schema discovery like [this](https://github.com/airbytehq/airbyte/blob/a381aa3727e7c7767440a6b937573c386a73e5de/airbyte-integrations/connectors/source-salesforce/source_salesforce/api.py#L351-L356)

Reply:
let's just call the parameter `name`? I like the idea of naming the client to help troubleshooting